tutorials/networking/websocket.rst

```diff
@@ -38,59 +38,32 @@ This example will show you how to create a WebSocket connection to a remote serv
 
     extends Node
 
-    # The URL we will connect to
-    export var websocket_url = "wss://libwebsockets.org"
-
     # Our WebSocketClient instance
-    var _client = WebSocketClient.new()
+    var _client = WebSocketPeer.new()
 
     func _ready():
-        # Connect base signals to get notified of connection open, close, and errors.
-        _client.connection_closed.connect(_closed)
-        _client.connection_error.connect(_closed)
-        _client.connection_established.connect(_connected)
-        # This signal is emitted when not using the Multiplayer API every time
-        # a full packet is received.
-        # Alternatively, you could check get_peer(1).get_available_packets() in a loop.
-        _client.data_received.connect(_on_data)
-
         # Initiate connection to the given URL.
-        var err = _client.connect_to_url(websocket_url, ["lws-mirror-protocol"])
+        # remote ip addr and port
+        var err = _client.connect_to_url("wss://addr:port")
         if err != OK:
             print("Unable to connect")
             set_process(false)
 
-    func _closed(was_clean = false):
-        # was_clean will tell you if the disconnection was correctly notified
-        # by the remote peer before closing the socket.
-        print("Closed, clean: ", was_clean)
-        set_process(false)
-
-    func _connected(proto = ""):
-        # This is called on connection, "proto" will be the selected WebSocket
-        # sub-protocol (which is optional)
-        print("Connected with protocol: ", proto)
-        # You MUST always use get_peer(1).put_packet to send data to server,
-        # and not put_packet directly when not using the MultiplayerAPI.
-        _client.get_peer(1).put_packet("Test packet".to_utf8())
-
-    func _on_data():
-        # Print the received packet, you MUST always use get_peer(1).get_packet
-        # to receive data from server, and not get_packet directly when not
-        # using the MultiplayerAPI.
-        print("Got data from server: ", _client.get_peer(1).get_packet().get_string_from_utf8())
-
     func _process(delta):
-        # Call this in _process or _physics_process. Data transfer, and signals
-        # emission will only happen when calling this function.
         _client.poll()
+        var state = socket.get_ready_state()
+    	if state == WebSocketPeer.STATE_OPEN:
+    		while socket.get_available_packet_count():
+    			print("data packet：", socket.get_packet())
+    	elif state == WebSocketPeer.STATE_CLOSING:
+    		# closing
+    		pass
+    	elif state == WebSocketPeer.STATE_CLOSED:
+    		var code = socket.get_close_code()
+    		var reason = socket.get_close_reason()
```

Comment:
```suggestion
        var state = _client.get_ready_state()
    	if state == WebSocketPeer.STATE_OPEN:
    		while _client.get_available_packet_count():
    			print("data packet：", _client.get_packet())
    	elif state == WebSocketPeer.STATE_CLOSING:
    		# closing
    		pass
    	elif state == WebSocketPeer.STATE_CLOSED:
    		var code = _client.get_close_code()
    		var reason = _client.get_close_reason()
```
`socket` is not defined as a variable here, you probably forgot to update when copying from the documentation of `WebSocketPeer`